db/db.go

```diff
@@ -0,0 +1,75 @@
+package db
+
+/*
+
+client<->entity relationship:
+
+get/set/delete accept the Path() argument from entity to perform key/value
+lookup. Get and set respectively manage Read() and Write() from entity.
+*/
+
+// Client provides a method of hooking into our data stores. It processes
+// entities (see below) and their pipeline between them and k/v stores such as
+// etcd and consul.
+type Client interface {
+	// Get the entity. Will copy the contents of the JSON into the entity for
+	// use.
+	Get(Entity) error
+	// Set the entity. Will persist the entity to etcd based on the path it
+	// provides. (See Path())
+	Set(Entity) error
```

Comment:
@erikh question: are there any stores which differentiate Add and Set ? Add - first time, Set - change(could be a subset of the entry)? If so, we need to design for them here.